galoy-client/src/client/mod.rs

```diff
@@ -0,0 +1,361 @@
+}
+
+pub(crate) struct StablesatsAuthToken {
+    pub inner: Option<String>,
+}
+
+#[derive(Debug)]
+pub struct OnchainAddress {
+    pub address: String,
+}
+
+impl GaloyClient {
+    pub async fn connect(config: GaloyClientConfig) -> Result<Self, GaloyClientError> {
+        let jwt = Self::login_jwt(config.clone()).await?;
+        let jwt = match jwt.inner {
+            Some(jwt) => jwt,
+            None => {
+                return Err(GaloyClientError::Authentication(
+                    "Empty authentication token".to_string(),
+                ))
+            }
+        };
+        let client = ReqwestClient::builder()
+            .default_headers(
+                std::iter::once((
+                    AUTHORIZATION,
+                    HeaderValue::from_str(&format!("Bearer {}", jwt)).unwrap(),
+                ))
+                .collect(),
+            )
+            .build()?;
+
+        Ok(Self {
+            client,
+            config,
+            jwt,
+        })
+    }
+
+    pub async fn authentication_code(
+        &self,
+    ) -> Result<StablesatsAuthenticationCode, GaloyClientError> {
+        let phone_number = stablesats_auth_code::Variables {
+            input: stablesats_auth_code::UserRequestAuthCodeInput {
+                phone: self.config.phone_number.clone(),
+            },
+        };
+        let response =
+            post_graphql::<StablesatsAuthCode, _>(&self.client, &self.config.api, phone_number)
+                .await?;
+        let response_data = response.data;
+
+        if let Some(response_data) = response_data {
+            StablesatsAuthenticationCode::try_from(response_data)?;
+        }
+        Err(GaloyClientError::UnknownResponse(
+            "Failed to parse response data".to_string(),
+        ))
+    }
+
+    async fn login_jwt(config: GaloyClientConfig) -> Result<StablesatsAuthToken, GaloyClientError> {
+        let variables = stablesats_user_login::Variables {
+            input: stablesats_user_login::UserLoginInput {
+                code: config.code.clone(),
+                phone: config.phone_number.clone(),
+            },
+        };
+
+        let client = ReqwestClient::new();
+
+        let response =
+            post_graphql::<StablesatsUserLogin, _>(&client, config.api, variables).await?;
+        if response.errors.is_some() {
+            if let Some(error) = response.errors {
+                return Err(GaloyClientError::GrapqQlApi(error[0].clone().message));
+            }
+        }
+
+        let response_data = response.data;
+        if response_data.is_none() {
+            return Err(GaloyClientError::UnknownResponse(
+                "Empty `data` in response".to_string(),
+            ));
+        }
+
+        let auth_token = match response_data {
+            Some(login_data) => StablesatsLogin::try_from(login_data)?.auth_token,
+            None => {
+                return Err(GaloyClientError::UnknownResponse(format!(
```

Comment:
What is the difference between `UnknownResponse` and `GraphQLApi`? You seem to be using both of those errors in the same way.

Reply:
Removed the redundant `UnknownResponse`